sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1994,6 +1996,20 @@ class Analyzer(
     }
   }
 
+  /**
+   * Set the seed for random number generation in Uuid expressions.
+   */
+  object ResolvedUuidExpressions extends Rule[LogicalPlan] {
+    private lazy val random = new Random()
+
+    override def apply(plan: LogicalPlan): LogicalPlan = plan.transformUp {
+      case p if p.resolved => p
+      case p => p transformExpressionsUp {
+        case Uuid(None) => Uuid(Some(random.nextLong()))
```

Comment:
shall we do the same thing for `Rand`?

Reply:
hmm, if we want to make it deterministic between re-tries of same query. I think we should do it. I can make a PR for it, WDYT?

Reply:
SGTM. We can even create a base trait for these random functions.

Reply:
I have checked, and actually `Rand` and `Randn` already have this behavior that they are deterministic between re-tries, though their random seeds are not determined at analysis but at constructing.

So I'm thinking should we do the same thing (random seed initialized at analysis) to `Rand` and `Randn`? Besides just to be consistent with `Uuid`, is any good reason to do this?

Reply:
One thing I think is, is any use case that we need to re-initialize random seed for `Rand`? Maybe streaming? For streaming query, I think `Rand` should use different random seed in each execution. For now, the random seed is initialized when constructing, even we re-analyze the query, it still uses same seed.

Reply:
what's the current behavior for rand in streaming?

Reply:
I created a PR #21854 which shows behavior of `Uuid` in streaming. I think rand should be the same.